src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import userEvent from '@testing-library/user-event'
```

Comment:
I commented throughout the tests, hope they're easy to follow. Most of my approach came from here: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library